packages/edit-site/src/components/start-template-options/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * WordPress dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useMemo } from '@wordpress/element';
+import { __experimentalBlockPatternsList as BlockPatternsList } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useAsyncList, useResizeObserver } from '@wordpress/compose';
+import { store as preferencesStore } from '@wordpress/preferences';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import { store as coreStore, useEntityBlockEditor } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+function useFallbackTemplateContent( slug, isCustom = false ) {
+	const [ templateContent, setTemplateContent ] = useState( '' );
+
+	useEffect( () => {
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/templates/lookup', {
+				slug,
+				is_custom: isCustom,
+				ignore_empty: true,
+			} ),
+		} ).then( ( { content } ) => setTemplateContent( content.raw ) );
+	}, [ slug ] );
+	return templateContent;
+}
+
+const START_BLANK_TITLE = __( 'Start blank' );
+
+function PatternSelection( { fallbackContent, onChoosePattern, postType } ) {
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const [ gridHeight, setGridHeight ] = useState( '320px' );
+	const [ , , onChange ] = useEntityBlockEditor( 'postType', postType );
+	const blockPatterns = useMemo(
+		() => [
+			{
+				name: 'fallback',
+				blocks: parse( fallbackContent ),
+				title: __( 'Fallback content' ),
+			},
+			{
+				name: 'start-blank',
+				blocks: parse(
+					'<!-- wp:paragraph --><p></p><!-- /wp:paragraph -->'
+				),
+				title: START_BLANK_TITLE,
+			},
+		],
+		[ fallbackContent ]
+	);
+	const shownBlockPatterns = useAsyncList( blockPatterns );
+	useEffect( () => {
+		const elementOffSetWidth = window?.document?.querySelector(
+			'.edit-site-start-template-options__pattern-container .block-editor-block-patterns-list__list-item'
+		)?.offsetWidth;
```

Comment:
What is this code about, is there a way to achieve the same without dom access?

Reply:
We want the templates in the grid to be 3/4 proportional, so we need to set height as a function of the width. I researched, and it seems there is no way to do that without dom access.

Reply:
Can this be some kind of "prop" to the BlockPreview or something like that. The block preview can use resize observers to do this properly. For instance, if you resize the page, this probably needs to be recomputed...

Reply:
> For instance, if you resize the page, this probably needs to be recomputed...
We are considering that because we are using a resize observer as the effect's trigger.

> Can this be some kind of "prop" to the BlockPreview or something like that.

I guess we can, but I think BlockPreview is supposed to be exposed, and I'm not sure if this behavior is needed elsewhere.

Reply:
I added a comment specifying what this is doing.

Reply:
Personally, this still bothers me, If we want the pattern list component to show a grid that we can customize, it should be a prop for that component

```
<BlockPatternList gridColumns={ 3 } />.
```

or something like that. Right now, it feels like we're just hacking CSS in addition to the JS hack to get the width of items which can change over time easily as we resize....

Reply:
Updated I used the CSS aspect ratio and remove all JS code related to the dimensions. Thank you for the tip.